controllers/workspace/provision/deployment.go

```diff
@@ -453,3 +574,11 @@ func checkContainerStatusForFailure(containerStatus *corev1.ContainerStatus) (ok
 	}
 	return true
 }
+
+func getEnvVar(key, value string) corev1.EnvVar {
+	envKey := DevWorkspaceEnvPrefix + "_" + (strings.ToUpper(strings.ReplaceAll(strings.ReplaceAll(key, ".", "_"), "-", "_")))
```

Comment:
Does Che Server appends their prefix as well?

If we replace . and -, should not we take care of other not allowed special characters, !@#$%... ?